### PR TITLE
Initial Work toward Webpack 5 with Svue

### PR DESCRIPTION
**Pull request by thadk**
- [x] Upgrade Svelte environment to latest using Webpack 4 successfully (fully running) 
- [x] Mitigate latest Svelte's accessibility and unused CSS warnings (this didn't work for our versions `https://github.com/sveltejs/language-tools/issues/650`)
- [x] Upgrade Docker image to Node 16 (stable), required for latest SASS, because bundled Python was 3.5 not 3.6. 
- [x] Upgrade and migrate SASS to latest, now non-compiled version https://sass-lang.com/documentation/breaking-changes/slash-div
- [ ] replace `node-sass` with `dart-sass`?
- [x] most basic typescript compiler support https://svelte.dev/blog/svelte-and-typescript 
- [ ] typescript in svelte https://github.com/baileyherbert/svelte-webpack-starter/blob/main/webpack.config.ts
- [x] upgrade jest and fix tests
- [x] upgrade babel
- [x] migrate webpack `devServer` from v3 to v4
- [x] migrate `svelte-preprocess` v3 to v4 https://github.com/sveltejs/svelte-preprocess/blob/main/docs/migration-guide.md
- [x] Initial env variables appearing in frontend again
- [ ] Production-ready env solution
- [ ] Try out mitigations for latest Svelte's accessibility and unused CSS warnings (some classes removed; some on:focus, on:blur added)
- [ ] Fix SpeedMeasure and MiniCss conflict with our setup https://github.com/stephencookdev/speed-measure-webpack-plugin/issues/149#issuecomment-833464523
- [x] Get Svue working (mangled variable issue?: crashes on very first $isLoaded if statement)
- [ ] Run svelte-check fewer times in deploy preview

### CODE REVIEW

**Comment by thadk** [comment]
Error from `$isLoading`
image

https://github.com/MuckRock/documentcloud-frontend/blob/2e7c22befc0e406f038b92bcbcc92d5a84eca723/src/Main.svelte#L145

File: src/common/Toast.svelte [hunk 1]
```diff
@@ -83,6 +83,7 @@
   on:click={() => dismiss(i)}
   on:mouseover={reset}
   on:mousemove={reset}
+  on:focus={() => null}
```
Comment (mitchelljkotler) [inline src/common/Toast.svelte:86]: What does this do?

File: src/pages/FlatPage.svelte [hunk 2]
```diff
@@ -250,7 +250,10 @@
         }
       }
 
-      :global(h5, h6) {
+      :global(h5) {
```
Comment (mitchelljkotler) [inline src/pages/FlatPage.svelte:253]: Why split these up?

File: webpack.base.config.js [hunk 3]
```diff
@@ -1,22 +1,45 @@
-const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+/* TODO: switch to imports instead of require https://github.com/baileyherbert/svelte-webpack-starter/blob/main/webpack.config.ts */
```
Comment (mitchelljkotler) [inline webpack.base.config.js:1]: Review

File: webpack.base.config.js [hunk 4]
```diff
@@ -57,7 +88,39 @@ module.exports = wrap({
           loader: "svelte-loader",
           options: {
             emitCss: prod,
-            hotReload: !prod,
+            hotReload: false, /* Turned off for newer versions */
+            hotOptions: { /* 
+              TK none of these alternated, non-default settings seem fix the
+               underlying problem with Svue
```
Comment (mitchelljkotler) [inline webpack.base.config.js:94]: What is the problem with Svue?

**Review by mitchelljkotler — COMMENTED** [review]
This should be based against `master` not `staging` - `staging` is not updated consistently.

**Pull request closed without merging**